powerstore/resource_volume_group.go

```diff
@@ -257,4 +298,50 @@ func (r resourceVolumeGroup) updateVolGroupState(volgroupState *models.Volumegro
 		volumeIDList = append(volumeIDList, types.StringValue(string(volumeID)))
 	}
 	volgroupState.VolumeIDs, _ = types.SetValue(types.StringType, volumeIDList)
+
+	//Update VolumeNames value from Plan to State
+	var volumeNames []string
+	for _, volumeName := range volGroupPlan.VolumeNames.Elements() {
+		volumeNames = append(volumeNames, strings.Trim(volumeName.String(), "\""))
+	}
+	volumeNameList := []attr.Value{}
+	for _, volumeName := range volumeNames {
+		volumeNameList = append(volumeNameList, types.StringValue(string(volumeName)))
+	}
+	volgroupState.VolumeNames, _ = types.SetValue(types.StringType, volumeNameList)
+
+	//Update ProtectionPolicyName value from Plan to State
+	volgroupState.ProtectionPolicyName = volGroupPlan.ProtectionPolicyName
+}
+
+func (r resourceVolumeGroup) fetchByName(plan *models.Volumegroup) (valid bool, err error) {
+	var volumeIds []string
+	if len(plan.VolumeIDs.Elements()) != 0 && len(plan.VolumeNames.Elements()) != 0 {
```

Comment:
The validation is added in the schema?

Reply:
yes, removed validation check from method.